osquery/core/sql.cpp

```diff
@@ -69,12 +69,23 @@ SQL::SQL(const std::string& q) {
   status_ = Status(code, getStringForSQLiteReturnCode(code));
 }
 
-QueryData SQL::rows() { return results_; }
+const QueryData& SQL::rows() { return results_; }
 
 bool SQL::ok() { return status_.ok(); }
 
+Status SQL::getStatus() { return status_; }
+
 std::string SQL::getMessageString() { return status_.toString(); }
 
+const std::string SQL::kHostColumnName = "_source_host";
+
+void SQL::annotateHostInfo() {
+  std::string hostname = getHostname();
```

Comment:
There's a way to use the machine's UUID instead of hostname (if you don't trust the DHCP/hostname: https://github.com/facebook/osquery/blob/master/osquery/scheduler/scheduler.cpp#L25-L35

This is exposed as `--host_identifier={uud, hostname}` as a config option or CLI switch, and `getHostIdentifier` will return the identifier based on the configured option.